src/ComposableController.ts

```diff
@@ -36,45 +22,22 @@ export class ComposableController extends BaseController<any, any> {
    * @param controllers - Map of names to controller instances
    * @param initialState - Initial state keyed by child controller name
    */
-  constructor(controllers: ControllerList = [], initialState?: any) {
-    super();
+  constructor(controllers: ControllerList) {
+    super(
+      undefined,
+      controllers.reduce((state, controller) => {
+        state[controller.name] = controller.state;
+        return state;
+      }, {} as any),
+    );
     this.initialize();
-    this.cachedState = initialState;
     this.controllers = controllers;
-    this.cachedState = undefined;
-  }
-
-  /**
-   * Get current list of child composed store instances
-   *
-   * @returns - List of names to controller instances
-   */
-  get controllers() {
-    return this.internalControllers;
-  }
-
-  /**
-   * Set new list of controller instances
-   *
-   * @param controllers - List of names to controller instsances
-   */
-  set controllers(controllers: ControllerList) {
-    this.internalControllers = controllers;
-    const initialState: any = {};
-    controllers.forEach((controller) => {
+    this.controllers.forEach((controller) => {
       const { name } = controller;
-      this.context[name] = controller;
-      controller.context = this.context;
-      this.cachedState?.[name] && controller.update(this.cachedState[name]);
```

Comment:
Apparently this line is how most controllers on mobile had their initial state set. I had wrongly assumed they were being passed initial state via their constructors.

This initial state setting has been re-implemented in mobile: https://github.com/MetaMask/metamask-mobile/pull/2416/commits/2b8023568161ad9d31b0e525f6c3318e71a436e5
Going forward we should set initial state in the constructor instead.